sdk/iot/core/tests/cmocka/main.c

```diff
@@ -24,9 +27,27 @@ void az_iot_sas_token_generate_sas_token_overflow_fails(void** state);
 void az_iot_sas_token_generate_succeeds(void** state);
 void az_iot_sas_token_generate_with_keyname_succeeds(void** state);
 
+/*
+ * Telemetry Unit Tests
```

Comment:
how out of control is this file going to get?

Reply:
honestly could be pretty hairy. I have additional thoughts about validating that all tests that are written are actually put here in this main. Inevitably tests are going to be written and someone will make an honest mistake and forget to copy over to here, test will never run, and we could have a test go unchecked.

Reply:
move this test definitions and adding tests to list to a  `az_test_definitions.h`
let's keep this file just to call tests like
```c
int main(void) { return cmocka_run_group_tests_name("az_iot_core", tests, NULL, NULL); }
```
See how it is on az_core...   

Well, that's just an option to avoid this file to keep growing and growing..
for az_core, I am trying to split things a little bit and I add some comments about how to add tests